src/test/java/com/revature/studyforce/flashcard/service/VoteServiceTest.java

```diff
@@ -0,0 +1,64 @@
+package com.revature.studyforce.flashcard.service;
+
+import com.revature.studyforce.flashcard.dto.VoteDTO;
+import com.revature.studyforce.flashcard.model.Answer;
+import com.revature.studyforce.flashcard.model.Flashcard;
+import com.revature.studyforce.flashcard.model.Vote;
+import com.revature.studyforce.flashcard.repository.AnswerRepository;
+import com.revature.studyforce.flashcard.repository.VoteRepository;
+import com.revature.studyforce.user.model.Authority;
+import com.revature.studyforce.user.model.User;
+import com.revature.studyforce.user.repository.UserRepository;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Test class for VoteService {@link VoteService}
+ * @author Nick Zimmerman
+ */
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@TestPropertySource(locations = "classpath:test-application.properties")
+class VoteServiceTest {
+
+    @MockBean
+    private UserRepository userRepo;
+    @MockBean
+    private AnswerRepository answerRepo;
+    @MockBean
+    private VoteRepository voteRepository;
+
+    @Autowired
+    private VoteService voteService;
+
+    @Test
+    void givenVoteDTOWithMatchingUserAndAnswer_whenAddVote_shouldPersistAndReturnVote() {
+        User u = new User(0,"jesus.christ@revature.com","password","Jesus","Christ",true,false,false, Authority.USER, Timestamp.valueOf(LocalDateTime.now()),Timestamp.valueOf(LocalDateTime.now()));
```

Comment:
Using outdated user model, please correct to user model with 1 name field and no password field